test/src/unittests/tonal/test_pitchsaliencefunction.py

```diff
@@ -0,0 +1,377 @@
+        """ FIXME
+        The intention here is to test if some peaks are correctly discarded, however, this test does not achieve this goal because there is only one input spectral peak.
+
+The magnitudeThreshold parameter defines the maximum allowed difference from the highest peak in dBs, so the input list of peaks should be of different magnitudes to see it working.
+
+For example, to make it simple, we can add a few peaks with lower magnitudes than 1 and use magnitudeThreshold = 0. Then only the peak with the magnitude 1 will be selected and the current expectedPitchSalience will be still a correct expected test output
+        """
+    def testTwoPeaksHarmonics(self):        
+        # Provide a 2 input peaks with a unit magnitude and validate
+        # that PSF is different depending on numberHarmonics configuration.               
+        freq_speaks = [55, 110] 
+        mag_speaks = [1, 1] 
+        outputLength  = 600        
+         # Length of the non-zero values for this Pitch Salience = 11
+        expectedPitchSalience = [1.0000000e+00, 9.7552824e-01, 9.0450847e-01, 7.9389262e-01, 6.5450847e-01,
+            5.0000000e-01, 3.4549147e-01, 2.0610739e-01, 9.5491491e-02, 2.4471754e-02,
+            3.7493994e-33, 0.0000000e+00]
+
+        calculatedPitchSalience1H = PitchSalienceFunction(numberHarmonics=1)(freq_speaks, mag_speaks)
+        self.assertEqual(len(calculatedPitchSalience1H), outputLength)               
+        calculatedPitchSalience20H = PitchSalienceFunction(numberHarmonics=20)(freq_speaks, mag_speaks)        
+        self.assertEqual(len(calculatedPitchSalience20H), outputLength)               
+        if calculatedPitchSalience1H.tolist()!=calculatedPitchSalience20H.tolist():
+        #  Return a TRUE result
+           self.assertEqual(1,1)
+        else:
+        #  Return a FALSE result       
+           self.assertEqual(1,0)        
+     
+    def testDuplicatePeaks(self):
+        # Provide multiple duplicate peaks at the reference frequency.    
+        freq_speaks = [55, 55, 55] 
+        mag_speaks = [1, 1, 1] 
+        outputLength  = 600        
+        # The same expectedPitchSalience from testSinglePeak test case
+        expectedPitchSalience = [3.0000000e+00, 2.9265847e+00, 2.7135253e+00, 2.3816779e+00, 1.9635254e+00,
+            1.5000000e+00, 1.0364745e+00, 6.1832219e-01, 2.8647447e-01, 7.3415264e-02, 1.1248198e-32]
+
+        # For 3 duplicate peaks, the expectedPitchSalience needs to be scaled by a factor of 3
+        arrayExpectedPitchSalience = 3*array(expectedPitchSalience)
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks, mag_speaks) 
+        # The next 10 values are decreasing in magnitude   
+        self.assertAlmostEqualVectorFixedPrecision(calculatedPitchSalience[:11], expectedPitchSalience, 7)
+        # Check remaining elements are zeros
+        self.assertEqualVector(calculatedPitchSalience[12:outputLength], zeros(outputLength-(12)))
```

Comment:
Let's rework this. Why not append the expected amount of zerots to the `expectedPitchSalience` vector and use a single `assertAlmostEqualVectorFixedPrecision`? It will be much cleaner.
```
expectedPitchSalience += [0] * <number of expected zeros>
```